pineappl_cli/src/evolve.rs

```diff
@@ -425,6 +425,7 @@ mod eko {
 fn evolve_grid(
     grid: &Grid,
     eko: &Path,
+    extra_eko: &Path,
```

Comment:
Here you can already start generalizing the code: replace `eko` and `extra_eko` with an `ekos: &[&Path]`. You can then use `match ekos.len() { 0 => /* Error */, 1 | 2 => {}, _ => unimplemented!() }` to catch the cases that don't make sense or aren't implemented yet.